drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -820,6 +803,9 @@ static inline void usbd_work_process_setup(struct nrf_usbd_ep_ctx *ep_ctx)
 	usbd_setup->wLength = nrf_usbd_setup_wlength_get(NRF_USBD);
 	ep_ctx->buf.len = sizeof(struct usb_setup_packet);
 
+	/* Copy setup packet to driver internal structure */
+	memcpy(&usbd_ctx.setup, usbd_setup, sizeof(struct usb_setup_packet));
```

Comment:
`usbd_ctx.setup = usbd_setup;` would be fine here. And probably such statement could be optimized well, as the structure is quite small.